cibyl/orchestrator.py

```diff
@@ -295,6 +295,7 @@ def run_query(self, system, start_level=1):
                     # if one source has provided the information, there is
                     # no need to query the rest
                     break
+        source_obj.ensure_teardown()
```

Comment:
This is still problematic because we cannot be sure that only one source has been used here. I think the better approach would be to loop over all sources and call the ensure_teardown method.

Reply:
Mmm. So in this case this method should be moved before the `break`? (https://github.com/rhos-infra/cibyl/blob/59188a50efe6ad837ed73dc1e9b43c444bc06d7f/cibyl/orchestrator.py#L297). That's the end of every iteration of the sources (end of the loop `for source_method, speed_score in source_methods:`).

Reply:
Not there, because the source might be used again later. I think it should be called after the run_query method and for all sources

Reply:
As we discussed in the call I've moved it.